controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -362,6 +362,13 @@ func (w *Workload) updateStaticPodCondition(ctx context.Context, machine *cluste
 		Name:      staticPodName(component, node.Name),
 	}
 
+	// If node ready is unknown there is a good chance that kubelet is not updating mirror pods, so we consider pod status
```

Comment:
nit: this block could go before declaring `podKey` block.

Reply:
done!